crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -0,0 +1,158 @@
+use crate::{
+    fuel_core_graphql_api::ports::worker::OffChainDatabaseTransaction,
+    graphql_api::storage::da_compression::{
+        DaCompressedBlocks,
+        DaCompressionTemporalRegistry,
+        DaCompressionTemporalRegistryEvictor,
+        DaCompressionTemporalRegistryIndex,
+    },
+};
+use fuel_core_compression::{
+    compress::compress,
+    ports::{
+        EvictorDb,
+        TemporalRegistry,
+        UtxoIdToPointer,
+    },
+    RegistryKeyspace,
+};
+use fuel_core_storage::{
+    not_found,
+    StorageAsMut,
+    StorageAsRef,
+};
+use fuel_core_types::{
+    blockchain::block::Block,
+    services::executor::Event,
+};
+use futures::FutureExt;
+
+/// Performs DA compression for a block and stores it in the database.
+pub fn da_compress_block<T>(
+    block: &Block,
+    events: &[Event],
+    transaction: &mut T,
+) -> anyhow::Result<()>
+where
+    T: OffChainDatabaseTransaction,
+{
+    let compressed = compress(CompressTx(transaction, events), block)
+        .now_or_never()
+        .expect("The current implementation resolved all futures instantly")?;
+
+    transaction
+        .storage_as_mut::<DaCompressedBlocks>()
+        .insert(&block.header().consensus().height, &compressed)?;
+
+    Ok(())
+}
+
+struct CompressTx<'a, Tx>(&'a mut Tx, &'a [Event]);
```

Comment:
[nit] I would prefer to have fields like `.block_db` and `.block_events` instead of `.0` and `.1`

Reply:
50d53d1424